test/Adapter.spec.js

```diff
@@ -142,4 +142,31 @@ describe('waitDevice', () => {
 
     return res
   })
+
+  test('clear intervals and timeouts after fail', async () => {
+    jest.useFakeTimers('legacy')
+
+    const adapter = new Adapter(dbus, 'hci0')
+
+    adapter.helper.children.mockResolvedValue([
+      'dev_11_11_11_11_11_11',
+      'dev_22_22_22_22_22_22',
+      'dev_33_33_33_33_33_33'
+    ])
+
+    const timeout = 500
+    const discoveryInterval = 100
+
+    const spyClearInterval = jest.spyOn(global, 'clearInterval')
+    const spyClearTimeout = jest.spyOn(global, 'clearInterval')
+
+    const waitDevicePromise = adapter.waitDevice('44:44:44:44:44:44', timeout, discoveryInterval)
+
+    jest.advanceTimersByTime(500)
```

Comment:
you can specify here the timeout variable that you declared

```suggestion
    jest.advanceTimersByTime(timeout)
```